.github/workflows/build.yaml

```diff
@@ -190,3 +190,10 @@ jobs:
         with:
           name: ${{matrix.platform}}-${{matrix.test-name}}-logs
           path: ${{matrix.platform}}-${{matrix.test-name}}-logs.tar.gz
+
+  all-passing:
+    name: All Required Tests Passed
+    needs: [tests, lr-tests, np-tests]
```

Comment:
Since `lr-tests` are not always run, does this work?

Reply:
I think so, because

> A default status check of `success()` is applied

and `success()` is defined as

> Returns `true` when none of the previous steps have failed or been canceled.

So `lr-tests` being _skipped_ means it is neither failed nor cancelled.

Similarly, `all-passing` will be skipped if dependent jobs fail, and afaict the branch protection rule does require the test to _pass_ -- skipped does not meet the requirement.

Reply:
welp, I guess not :joy:

Frustratingly I checked a simple workflow in a private repo beforehand but apparently some combination I didn't account for.

Reply:
Lol, I noticed the test run and saw it passed, so approved it.

Reply:
> Similarly, all-passing will be skipped if dependent jobs fail, and afaict the branch protection rule does require the test to pass -- skipped does not meet the requirement.

and bizarrely that seems false as well